### PR TITLE
Added kl_div frontend

### LINKED ISSUE #9183: kl_div

### PR DESCRIPTION
**Pull request by zaRizk7**
I was trying to add test cases and ran pytest on the `test_loss_functions.py` file, it seems when i try to run it certain functions in the helper like `array_or_none` does not exists which gave me an error while testing. I tried it on my other files for empirical testing and the results between original functionals for `kl_div` should work as expected.

closes #9183

### CODE REVIEW

**Comment by zaRizk7** [comment]
Also, may I have this request linked to my issue? I can't seem to find the settings icon to link to it by myself. My issue is #9183

**Comment by zaRizk7** [comment]
@juliagsy may I ask for my current code progress to be reviewed? If there's like any additional bug I could fix as currently, the test case for the current kl_div I have pushed have been successful as of ight now.

**Comment by juliagsy** [comment]
Hey! Thanks for contributing! I'll review within the next 2 days! Meanwhile, could you please resolve the conflicts in the test file? Thanks!

File: ivy/functional/frontends/torch/nn/functional/loss_functions.py [hunk 1]
```diff
@@ -208,43 +208,80 @@ def l1_loss(
...
+    size = ivy.shape(input)
+    
+    def loss_fn():
+        if log_target:
+            return ivy.exp(target) * (target - input)
+        return target * (ivy.log(target) - input)
+    
+    def batchmean(x):
+        if not reduce:
+            return x / size[0]
```
Comment (juliagsy) [inline ivy/functional/frontends/torch/nn/functional/loss_functions.py:270]: this should accommodate for cases where arrays are 0-dim.

when an array is 0-dim, the shape is a tuple of (), meaning length 0. the test fails in this case

File: ivy_tests/test_ivy/test_frontends/test_torch/test_loss_functions.py [hunk 2]
```diff
@@ -158,103 +158,103 @@ def test_torch_binary_cross_entropy(
...
-        on_device=on_device,
-        input=pred[0],
-        target=true[0],
-        weight=weight[0],
-        size_average=size_average,
-        reduce=reduce,
-        reduction=reduction,
-        pos_weight=pos_weight[0],
-    )
+# @handle_frontend_test(
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_loss_functions.py:161]: is this test commented out originally? if not please avoid changes outside your selected function, thank you!

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey there! Great work so far! Some changes related to test failures are needed, please request my review once again when you're done, thanks!

File: ivy/functional/frontends/torch/nn/functional/loss_functions.py (hunk 1 above)
Comment (zaRizk7) [reply ivy/functional/frontends/torch/nn/functional/loss_functions.py:273]: I've fixed this by adding a handler for the size length.

Comment (zaRizk7) [reply ivy/functional/frontends/torch/nn/functional/loss_functions.py:273]: I also updated the test case to handle scalar now.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_loss_functions.py (hunk 2 above)
Comment (zaRizk7) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_loss_functions.py]: I fix the conflict with the latest master which fixes the test error for bce.

Comment (zaRizk7) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_loss_functions.py]: I've fixed this one by resolving the conflicts with the latest master branch. Turns out the testing bug in BCE loss have been fixed. @juliagsy I would like my latest pushes be reviewed.

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey there! Thanks for the changes!

The tests for `kl_div` are still failing, you may check the workflow for the failures, or you may run the tests locally by executing:
```
python3 -m pytest ivy_tests/test_ivy/test_frontends/test_torch/test_loss_functions.py::test_torch_kl_div
```

You should ensure that the tests for `kl_div` are passing for all backends.
Thanks!

**Review by zaRizk7 — COMMENTED** [review]
@juliagsy turns out the error I got is due to I forgot to add unsupported decorator for half precision. It should work fine now as it has passed all of the tests in my machine.

**Review by juliagsy — APPROVED** [review]
Thanks for contributing! will merge in soon!

**Pull request merged**